File: testdata/jennies/rawtypes/variant_panelcfg_full/PHPRawTypes/src/VariantPanelcfgFull/Options.php
<?php

namespace Grafana\Foundation\VariantPanelcfgFull;

class Options implements \JsonSerializable
{
    public string $timeseriesOption;

    /**
     * @param string|null $timeseriesOption
     */
    public function __construct(?string $timeseriesOption = null)
    {
        $this->timeseriesOption = $timeseriesOption ?: "";
    }

    public function equals(mixed $other): bool
    {
        if (!($other instanceof self)) {
            return false;
        }
    
        if ($this->timeseriesOption !== $other->timeseriesOption) {
            return false;
        }
    
        return true;
    }

    /**
     * @param array<string, mixed> $inputData
     */
    public static function fromArray(array $inputData): self
    {
        /** @var array{timeseries_option?: string} $inputData */
        $data = $inputData;
        return new self(
            timeseriesOption: $data["timeseries_option"] ?? null,
        );
    }

    /**
     * @return mixed
     */
    public function jsonSerialize(): mixed
    {
        $data = new \stdClass;
        $data->timeseries_option = $this->timeseriesOption;
        return $data;
    }
}

File: testdata/jennies/rawtypes/variant_panelcfg_full/JavaRawTypes/variant_panelcfg_full/Options.java
package variant_panelcfg_full;

import java.util.Objects;

public class Options {
    public String timeseriesOption;
    public Options() {
        this.timeseriesOption = "";
    }
    public Options(String timeseriesOption) {
        this.timeseriesOption = timeseriesOption;
    }

    @Override
    public boolean equals(Object other) {
        if (this == other) return true;
        if (!(other instanceof Options)) return false;
        Options o = (Options) other;
        if (!Objects.equals(this.timeseriesOption, o.timeseriesOption)) return false;
        return true;
    }

    @Override
    public int hashCode() {
        return Objects.hash(this.timeseriesOption);
    }
}
